icepyx/core/granules.py

```diff
@@ -43,6 +44,55 @@ def gran_IDs(grans):
 
     return gran_ids
 
+def gran_cycles(grans):
+    """
+    Returns a list of the available orbital cycles for the granules
+    """
+    assert len(grans) > 0, "Your data object has no granules associated with it"
+    # regular expression for extracting parameters from file names
+    rx = re.compile('(ATL\d{2})(-\d{2})?_(\d{4})(\d{2})(\d{2})(\d{2})(\d{2})'
+           '(\d{2})_(\d{4})(\d{2})(\d{2})_(\d{3})_(\d{2})(.*?).(.*?)$')
+    gran_cycles = []
+    for gran in grans:
+        # PRD: ICESat-2 product
+        # HEM: Sea Ice Hemisphere flag
+        # YY,MM,DD,HH,MN,SS: Year, Month, Day, Hour, Minute, Second
+        # TRK: Reference Ground Track (RGT)
+        # CYCL: Orbital Cycle
+        # GRAN: Granule region (1-14)
+        # RL: Data Release
+        # VERS: Product Version
+        # AUX: Auxiliary flags
+        # SFX: Suffix (h5)
+        PRD,HEM,YY,MM,DD,HH,MN,SS,TRK,CYCL,GRAN,RL,VERS,AUX,SFX = \
+            rx.findall(gran["producer_granule_id"]).pop()
+        gran_cycles.append(CYCL)
+    return gran_cycles
+
+def gran_RGTs(grans):
+    """
+    Returns a list of the available Reference Ground Tracks (RGTs) for the granules
+    """
+    assert len(grans) > 0, "Your data object has no granules associated with it"
+    # regular expression for extracting parameters from file names
+    rx = re.compile('(ATL\d{2})(-\d{2})?_(\d{4})(\d{2})(\d{2})(\d{2})(\d{2})'
+           '(\d{2})_(\d{4})(\d{2})(\d{2})_(\d{3})_(\d{2})(.*?).(.*?)$')
+    gran_tracks = []
+    for gran in grans:
+        # PRD: ICESat-2 product
+        # HEM: Sea Ice Hemisphere flag
+        # YY,MM,DD,HH,MN,SS: Year, Month, Day, Hour, Minute, Second
+        # TRK: Reference Ground Track (RGT)
+        # CYCL: Orbital Cycle
+        # GRAN: Granule region (1-14)
+        # RL: Data Release
+        # VERS: Product Version
+        # AUX: Auxiliary flags
+        # SFX: Suffix (h5)
+        PRD,HEM,YY,MM,DD,HH,MN,SS,TRK,CYCL,GRAN,RL,VERS,AUX,SFX = \
+            rx.findall(gran["producer_granule_id"]).pop()
+        gran_tracks.append(TRK)
+    return gran_tracks
```

Comment:
Best practices question (I have no idea what the correct answer is, if there even is one): it seems like `gran_IDs()`, `gran_cycles()`, and `gran_RGTs()` all do almost the same thing. Should we be thinking about turning them into one method? I'm just thinking that if you're querying enough of the archive you could potentially see performance improvements if you only iterate through the granule list once.

Reply:
They do almost exactly the same thing.  A few commits ago they were a single function with kwargs.  I'll put it back that way :+1:

Reply:
Cool - thanks for being patient with my learning curve!